### ISSUE TITLE
New CCD <-> CAN board coming soon

### ISSUE THREAD

**Issue by csouers** [csouers/chrysler-ccd#6]
Will have bidirectional support and support Arduino pro mini and Particle devices.

The board will have a OBD2 header for connecting devices like the comma.ai panda for can sniffing and sending. The ISO/SCI lines will also be routed to OBD2 in their necessary spots to match the newer pinout spec (not sure if they move) if possible.

**Comment by causalloop** [comment]
Will this mean communication with older pre OBD2 vehicles using CCD - IE my 94 grand cherokee etc.?  Maybe I'm picturing this in my head wrong - is this creating an obd 2 port allowing for things to plug into it - or is this something that plugs into an obd2 port?

**Comment by csouers** [comment]
Technically, it will be possible. Note that this will not enable apps like Torque or Car Scanner to work with OBD1 cars. This is essentially just an adapter to go to CAN. The comma ai panda can read can busses, and when paired with cabana.comma.ai and your computer, you can view in real time the messages on the bus. Particle also has some potential here for cellular/cloud control of the bus. I've sent door unlock and lock commands over CCD from my phone. It's just a matter of sorting out the hardware and software.

This will either plug into the existing obd2 port or have some sort of a connection to run the ccd lines into (for OBD1 cars).

**Comment by causalloop** [comment]
I just want to get some sensor data without re-inventing the wheel..  as
long as there's adequate examples for whatever arduion or particle (yay!),
I'm a happy camper.  I've got an arduino in there already, but its just
connected to a couple of temp sensors I installed myself.

**Comment by csouers** [comment]
What data are you wanting off of the bus?

**Comment by causalloop** [comment]
Depends on whats available, but I'd like to get whatever temp data there
is, timing, vacuum pressure, fuel pressure, something that might help
indicate knock count etc.

**Comment by csouers** [comment]
yeah, i don't think any of that is on the bus (remember, this is not can. bandwidth and device counts are very limited!) . take a look in the google sheet in the readme. a majority of stuff is broken out there. At best you'll get coolant temp, maybe not on your 94. Does your ZJ have 2 coolant sensors (one for the pcm other for the gauge)?

**Comment by causalloop** [comment]
I picked up a snapon tool and can read quite a bit of data, which is
impressive considering it looks like it was made in like 1995 - ie huge
comparitively.  I considered leveraging it, but that just seemed like a
charlie foxtrot.

2 sensors for coolant - I actually pulled the one for the gauge and
replaced it with a digital sensor hooked up to my arduino so I could
actually get numbers (was having a hard time tracking down a coolant
problem for a while)

**Comment by csouers** [comment]
The snap on is reading from SCI. Totally different animal.
